library/azure_rm_applicationsecuritygroup.py

```diff
@@ -0,0 +1,250 @@
+    """Configuration class for an Azure RM Application Security Group resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.location = None
+        self.name = None
+        self.tags = None
+
+        self.state = None
+
+        self.results = dict(changed=False)
+
+        self.to_do = Actions.NoAction
+
+        super(AzureRMApplicationSecurityGroup, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                              supports_check_mode=True,
+                                                              supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+
+        resource_group = self.get_resource_group(self.resource_group)
+
+        if not self.location:
+            self.location = resource_group.location
+
+        old_response = self.get_applicationsecuritygroup()
+
+        if not old_response:
+            self.log("Application Security Group instance doesn't exist")
+            if self.state == 'present':
+                self.to_do = Actions.CreateOrUpdate
+            else:
+                self.log("Old instance didn't exist")
+        else:
+            self.log("Application Security Group instance already exists")
+            if self.state == 'present':
+                if self.check_update(old_response):
+                    self.to_do = Actions.CreateOrUpdate
+
+                update_tags, self.tags = self.update_tags(old_response.get('tags', None))
+                if update_tags:
+                    self.to_do = Actions.CreateOrUpdate
+
+            elif self.state == 'absent':
+                self.to_do = Actions.Delete
+
+        if self.to_do == Actions.CreateOrUpdate:
+            self.log("Need to Create / Update the Application Security Group instance")
+            self.results['changed'] = True
+
+            if self.check_mode:
+                return self.results
+
+            response = self.create_update_applicationsecuritygroup()
+            self.results['id'] = response['id']
+
+        elif self.to_do == Actions.Delete:
+            self.log("Delete Application Security Group instance")
+            self.results['changed'] = True
+            self.results['id'] = old_response.get('id', None)
```

Comment:
I am not sure about this.
I mean if instance is deleted, we should not return "id" any longer as it's no longer valid.
I think we have never considered that though....

Reply:
this is good point. let me remove this.